desktop/src/main.rs

```diff
@@ -216,7 +223,21 @@ fn run_player(opt: Opt) -> Result<(), Box<dyn std::error::Error>> {
             .with_max_inner_size(LogicalSize::new(i16::MAX, i16::MAX))
             .build(&event_loop)?,
     );
-    window.set_inner_size(movie_size);
+
+    let movie_size =
+        LogicalSize::new(movie.width(), movie.height()).to_physical(window.scale_factor());
+    let window_width = opt.width.unwrap_or(
+        (f64::from(movie_size.width)
+            * (f64::from(opt.height.unwrap_or(movie_size.height)) / f64::from(movie_size.height)))
+        .round() as u32,
+    );
+    let window_height = opt.height.unwrap_or(
+        (f64::from(movie_size.height)
+            * (f64::from(opt.width.unwrap_or(movie_size.width)) / f64::from(movie_size.width)))
+        .round() as u32,
+    );
+    let window_size = PhysicalSize::new(window_width, window_height);
+    window.set_inner_size(window_size);
```

Comment:
Let's make `movie_size` use `f64`:
`LogicalSize::new(movie.width(), movie.height()).cast::<f64>().to_physical(window.scale_factor())`
Then all of the `f64::from` goes away. winit takes care of the rounding when we call `set_inner_size`